deployment.rb

```diff
@@ -285,6 +285,20 @@ def method_missing(*args)
   def with_default(default_value)
     DelegateWithDefault.new(self, default_value)
   end
+
+  # When running on Chef Server, use Search via knife CLI to fetch a dynamic list of app-server front-ends,
+  # appending to the static list already provided by configuration files.
+  def app_servers
+    return super unless CDO.chef_managed
+    require 'cdo/rake_utils'
+    servers = RakeUtils.with_bundle_dir(cookbooks_dir) do
+      knife_cmd = "knife search node 'roles:front-end AND chef_environment:#{rack_env}' --format json --attribute cloud_v2"
```

Comment:
Yeah, I think correct behavior if chef is currently unavailable would be toraise the exception rather than return cached results, since the deploy would fail further down the process if chef is unavailable anyway.